services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_tasks/_core.py

```diff
@@ -265,3 +267,19 @@ async def delete_tasks_from_project(self, project_id: ProjectID) -> None:
             await conn.execute(
                 sa.delete(comp_tasks).where(comp_tasks.c.project_id == f"{project_id}")
             )
+
+    async def get_outputs_from_tasks(
+        self, project_id: ProjectID, node_ids: set[NodeID]
+    ) -> dict[NodeID, dict[str, Any]]:
+        selection = list(map(str, node_ids))
+        query = sa.select(comp_tasks.c.node_id, comp_tasks.c.outputs).where(
+            (comp_tasks.c.project_id == f"{project_id}")
+            & (comp_tasks.c.node_id.in_(selection))
+        )
+        async with self.db_engine.acquire() as conn:
+            result: ResultProxy = await conn.execute(query)
+            rows: list[RowProxy] | None = await result.fetchall()
```

Comment:
here I am always a bit puzzled, whether using fetchall vs async for is the rigt way. I guess in this case you never have millions of entries. nevertheless I think sqlalchemy ususally shows the async for type which is probably less memory intensive.
anyway.